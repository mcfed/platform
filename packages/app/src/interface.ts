export interface User {
  id?: number;
  accountName?: string;
  level?: string;
  needSetPd?: boolean;
  token?: string;
  userName?: string;
}

export interface ApiResData<T> {
  code: number;
  msg: string;
  data: T;
}

export interface RouteItem<T = any, K = string> {
  id?: string | number;
  name: K;
  key?: K;
  icon?: string;
  path: string;
  component?: any;
  visiable?: boolean;
  routes?: RouteItem<T, K>[];
}

export type RouterConfig<T = any, K = string> = RouteItem<T, K>[];
