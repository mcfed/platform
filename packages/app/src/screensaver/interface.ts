export interface IScreenSaver {
  checkExpires(date: Date): Boolean;
  lockScreen(): void;
  delayLock(): void;
  calcDelay(): Date;
}

export interface ILocker {
  locked(): void;
  verify(): void;
  unlocker(): void;
  render(): any;
}
