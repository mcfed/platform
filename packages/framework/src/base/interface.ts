import {IRFormProps, IRListProps} from '@mcfed/crud';

export interface ApiResData<T> {
  code: number;
  msg: string;
  data: T;
}

interface DictItem {
  label: string;
  value: string;
}

type Dicts = {
  (key: string, value: string): string;
  (key: string): DictItem[];
};

interface BasicPageProps {
  dicts: Dicts;
  appReducer: any;
}

export interface BaseFormViewProps extends IRFormProps, BasicPageProps {
  spins: (action: Function | string) => boolean;
  querys: (action: Function | string) => boolean;
  locale: (type: string, vars?: any) => string;
}

export interface BaseListViewProps extends IRListProps, BasicPageProps {
  spins: (action: Function | string) => boolean;
  querys: (action: Function | string) => boolean;
  locale: (type: string, vars?: any) => string;
}
