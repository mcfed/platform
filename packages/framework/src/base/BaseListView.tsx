import React from 'react';
import {IRListState, IParams, PK, RListPage} from '@mcfed/crud';
import {BaseListViewProps} from './interface';
//@ts-ignore
import {TableProps, PaginationConfig} from 'antd/lib/table';
// import styles from '../styles/style.module.css'

export default abstract class BaseListView<
  P extends BaseListViewProps,
  S extends IRListState
> extends RListPage<P, S> {
  abstract searchParams(): IParams<any>;
  abstract handleFilter(value: any): void;
  abstract componentDidMount(): void;
  abstract render(): React.ReactNode;
  onChange(pagination: PaginationConfig, filters: any, sorter: any) {
    const {current, pageSize} = pagination;
    var object = Object.assign(
      {},
      this.searchParams(),
      {page: current, size: pageSize},
      filters,
      sorter
    );
    this.handleFilter(object);
  }
  renderSearchBar() {
    return <div>sdfsfs</div>;
  }
  mergeTableConfig(config: TableProps<any>) {
    return Object.assign(
      {
        size: 'middle',
        onChange: this.onChange.bind(this),
        pagination: {
          showQuickJumper: true,
          showSizeChanger: true,
          pageSizeOptions: ['10', '20', '50', '100'],
          size: 'small',
          showTotal: (total: number, range: number[]) =>
            `共 ${total} 条，当前第 ${range[0]} 至 第 ${range[1]} 条`
        }
      },
      config,
      config?.rowSelection === null
        ? {}
        : {
            rowSelection: {
              onChange: this.onSelectChange?.bind(this),
              selectedRowKeys: this.state.selectedRowKeys,
              ...config.rowSelection
            }
          }
    );
  }
}
