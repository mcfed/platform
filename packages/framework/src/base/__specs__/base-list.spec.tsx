import React from 'react';
import {shallow, ShallowRendererProps} from 'enzyme';
import {IRListState} from '@mcfed/crud';
import BaseListView from '../BaseListView';
import {BaseListViewProps} from '../interface';

class TestListView extends BaseListView<BaseListViewProps, IRListState> {
  componentDidMount() {}
  handleFilter() {}
  searchParams() {
    return {};
  }
  render() {
    return 123;
  }
}

function setup(props: any, options?: ShallowRendererProps) {
  const defaultProps = {};
  const wrapper = shallow<TestListView>(
    <TestListView {...Object.assign({}, defaultProps, props)} />,
    options
  );
  return {
    wrapper
  };
}

describe('方法测试全覆盖', () => {
  const props = {};

  it('handleFilter', () => {
    const {wrapper} = setup(props, {disableLifecycleMethods: false});
    const instance = wrapper.instance();
    const handleFilterSpy = jest
      .spyOn(instance, 'handleFilter')
      .mockImplementation(() => {});
    const searchParamsSpy = jest
      .spyOn(instance, 'searchParams')
      .mockImplementation(() => ({params: 'test'}));
    instance.onChange({pageSize: 1, current: 2}, null, null);
    expect(searchParamsSpy).toHaveBeenCalled();
    expect(handleFilterSpy).toHaveBeenCalledWith({
      params: 'test',
      page: 2,
      size: 1
    });
  });

  it('mergeTableConfig|空参', () => {
    const {wrapper} = setup(props, {disableLifecycleMethods: false});
    const instance = wrapper.instance();
    expect(instance.mergeTableConfig({})).toMatchSnapshot();
  });

  it('mergeTableConfig|存在rowSelection属性', () => {
    const {wrapper} = setup(
      {
        rowSelection: {}
      },
      {disableLifecycleMethods: false}
    );
    const instance = wrapper.instance();
    expect(instance.mergeTableConfig({})).toMatchSnapshot();
  });

  it('mergeTableConfig|返回的showTotal', () => {
    const {wrapper} = setup(props, {disableLifecycleMethods: false});
    const instance = wrapper.instance();
    const showTotal = instance.mergeTableConfig({})?.pagination?.showTotal;
    expect(showTotal(15, [11, 15])).toMatchSnapshot();
  });
});
