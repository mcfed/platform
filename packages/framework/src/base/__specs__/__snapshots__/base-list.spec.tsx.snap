// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`方法测试全覆盖 mergeTableConfig|存在rowSelection属性 1`] = `
Object {
  "onChange": [Function],
  "pagination": Object {
    "pageSizeOptions": Array [
      "10",
      "20",
      "50",
      "100",
    ],
    "showQuickJumper": true,
    "showSizeChanger": true,
    "showTotal": [Function],
    "size": "small",
  },
  "rowSelection": Object {
    "onChange": [Function],
    "selectedRowKeys": Array [],
  },
  "size": "middle",
  "style": Object {
    "width": "100%",
  },
}
`;

exports[`方法测试全覆盖 mergeTableConfig|空参 1`] = `
Object {
  "onChange": [Function],
  "pagination": Object {
    "pageSizeOptions": Array [
      "10",
      "20",
      "50",
      "100",
    ],
    "showQuickJumper": true,
    "showSizeChanger": true,
    "showTotal": [Function],
    "size": "small",
  },
  "rowSelection": Object {
    "onChange": [Function],
    "selectedRowKeys": Array [],
  },
  "size": "middle",
  "style": Object {
    "width": "100%",
  },
}
`;

exports[`方法测试全覆盖 mergeTableConfig|返回的showTotal 1`] = `"共 15 条，当前第 11 至 第 15 条"`;
